LoRaEngine/LoraKeysManagerFacade/LoRaDevAddrCache.cs

```diff
@@ -203,18 +206,17 @@ private async Task PerformFullReload(IDeviceRegistryManager registryManager)
         /// </summary>
         private async Task PerformDeltaReload(IDeviceRegistryManager registryManager)
         {
-            // if the value is null (first call), we take five minutes before this call
-            var lastUpdate = this.cacheStore.StringGet(LastDeltaUpdateKeyValue) ?? DateTime.UtcNow.AddMinutes(-5).ToString(LoraKeysManagerFacadeConstants.RoundTripDateTimeStringFormat, CultureInfo.InvariantCulture);
+            // if the value is null (first call), we take updates from one hour before this call
+            var lastUpdate = this.cacheStore.StringGet(LastDeltaUpdateKeyValue) ?? DateTime.UtcNow.AddHours(-1).ToString(LoraKeysManagerFacadeConstants.RoundTripDateTimeStringFormat, CultureInfo.InvariantCulture);
             var query = $"SELECT * FROM devices where properties.desired.$metadata.$lastUpdated >= '{lastUpdate}' OR properties.reported.$metadata.DevAddr.$lastUpdated >= '{lastUpdate}'";
-            var devAddrCacheInfos = await GetDeviceTwinsFromIotHub(registryManager, query);
+            var devAddrCacheInfos = await GetDeviceTwinsFromIotHub(registryManager, query, DateTime.Parse(lastUpdate, CultureInfo.InvariantCulture));
             BulkSaveDevAddrCache(devAddrCacheInfos, false);
         }
 
-        private async Task<List<DevAddrCacheInfo>> GetDeviceTwinsFromIotHub(IDeviceRegistryManager registryManager, string inputQuery)
+        private async Task<List<DevAddrCacheInfo>> GetDeviceTwinsFromIotHub(IDeviceRegistryManager registryManager, string inputQuery, DateTimeOffset? lastDeltaUpdateFromCache)
         {
             var query = registryManager.CreateQuery(inputQuery);
-            var lastQueryTs = DateTime.UtcNow.AddSeconds(-10); // account for some clock drift
-            _ = this.cacheStore.StringSet(LastDeltaUpdateKeyValue, lastQueryTs.ToString(LoraKeysManagerFacadeConstants.RoundTripDateTimeStringFormat, CultureInfo.InvariantCulture), TimeSpan.FromDays(1));
+            var isFullReload = lastDeltaUpdateFromCache is null;
```

Comment:
general question: we know that the changes are propagated delayed. Are the timestamps synchronized when the changes propagate, or when they were made? If it is when they were made, and it could happen that they are not propagated all together, wouldn't there be a chance that we could assume we last updated > than a not yet propagated change?

Reply:
The timestamps I could see are about update time and not propagated time.

As far as I understood some properties in the registry are eventually consistent, therefore it could happen that some changes were not propagated.

It's hard to make a call here, not sure whether this is the optimal solution, but it's definitely better than what we had before.

Any suggestion?

Reply:
To mitigate, we could add an overlap of let's say 10min to the last used timestamp. That would mean, we will always get the ones that were updated last though. However, with the changes on proactively adding the entry when joining, I think the risk that we will miss an item is very low. Hence, We should document the possibility, but argue that we believe it is a managable risk.

Reply:
Comment added with e9a99ee8a6b364c70bbddf9d6c47dcd2e49cdbfb